crates/gpui/src/keymap/context.rs

```diff
@@ -1,14 +1,13 @@
 use crate::SharedString;
 use anyhow::{anyhow, Result};
-use smallvec::SmallVec;
 use std::fmt;
 
 /// A datastructure for resolving whether an action should be dispatched
 /// at this point in the element tree. Contains a set of identifiers
 /// and/or key value pairs representing the current context for the
 /// keymap.
 #[derive(Clone, Default, Eq, PartialEq, Hash)]
-pub struct KeyContext(SmallVec<[ContextEntry; 1]>);
+pub struct KeyContext(Vec<ContextEntry>);
```

Comment:
We could even opt for `Box<[ContextEntry]>` if you want to reduce the stack size of KeyContext to 16 bytes (and don't need mutability).

Reply:
Good point!  Probably possible, but current interface relies a lot on mutability